src/components/Tabs/Tabs.tsx

```diff
@@ -0,0 +1,195 @@
+        return () => {
+            window.removeEventListener("resize", checkIfOverflowing);
+        };
+    }, [checkIfOverflowing]);
+
+    const tabNavRef = useRef<HTMLDivElement | null>(null);
+
+    return (
+        <div data-test-id="tabs">
+            <div className="tw-flex tw-relative tw-border-b tw-border-grey-20">
+                <div
+                    ref={tabNavRef}
+                    className={merge([
+                        "tw-overflow-hidden tw-flex-shrink-0 tw-h-full tw-w-full tw-flex tw-justify-start",
+                        paddingMap[paddingX],
+                        size === TabSize.Small ? "tw-text-sm" : "tw-text-md",
+                    ])}
+                >
+                    {tabs.map((tab, index) => {
+                        return (
+                            <button
+                                data-test-id="tab-item"
+                                className={merge([
+                                    "tw-group tw-relative tw-mx-0 tw-pb-5 tw-pt-2 tw-px-2 tw-w-max tw-h-10 tw-cursor-pointer tw-flex tw-items-center tw-justify-center tw-whitespace-nowrap",
+                                    tab.disabled && "tw-text-text-disabled",
+                                    tab.id === activeItemId && "tw-font-medium",
+                                    index >= tabIndexLimit && "tw-invisible",
+                                ])}
+                                key={tab.id}
+                                onClick={() => !tab.disabled && onChange(tab.id)}
+                            >
+                                {tab.decorator}
+                                <span className="tw-mr-1 tw-ml-1.5">{tab.label}</span>
+                                {tab.badge && (
+                                    <Badge disabled={tab.disabled} style={tab.badge.style}>
+                                        {tab.badge.children}
+                                    </Badge>
+                                )}
+                                {tab.id === activeItemId && (
+                                    <motion.div
+                                        data-test-id="tab-active-highlight"
+                                        layoutId="underline"
+                                        className="tw-absolute tw-h-1 tw-bg-violet-60 tw-rounded-t-x-large tw-w-full tw-bottom-0"
+                                    />
+                                )}
+                                {tab.id !== activeItemId && !tab.disabled && (
+                                    <div className="group-hover:tw-absolute group-hover:tw-h-1 group-hover:tw-bg-grey-20 group-hover:tw-rounded-t-x-large group-hover:tw-w-full group-hover:tw-bottom-0" />
+                                )}
+                            </button>
+                        );
+                    })}
+                </div>
+                {isOverflowing && (
+                    // Remove rotate class when new Dots icon is available
```

Comment:
```suggestion
```
You can create a ClickUp ticket to follow up but it is also fine how it is right now